docs/API.md

```diff
@@ -332,71 +321,8 @@ scope.activate(outerSpan, () => {
 })
 ```
 
-<h6>Promise binding</h6>
```

Comment:
Should this be deleting the promise binding docs too? Seems to still have promise binding in the TS types. 🤔

Reply:
As described above, I went ahead and removed the docs for that since it should no longer be needed. We can remove it proper in a future major when it's no longer used internally.

Reply:
Ah, so the types need to stay for now, basically?

Reply:
Yes and no, we could also remove promises right away, it's unlikely to be used since users would already by either using promise libraries or thenables that we patch, or native promises that are already bound properly. To be honest thinking more about it I don't see why we couldn't just remove both right away. For some reason I thought this is still used internally but that isn't the case.

Reply:
I'll open another PR to remove promise binding and `unbind` because these should not be used and if they are the alternatives are simple enough.